instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java

```diff
@@ -0,0 +1,173 @@
+
+  public static Stream<Arguments> headerKeys() {
+    return Arrays.stream(
+            new String[] {
+              "otel.exporter.otlp.traces.headers",
+              "otel.exporter.otlp.metrics.headers",
+              "otel.exporter.otlp.logs.headers",
+              "otel.exporter.otlp.headers",
+            })
+        .map(Arguments::of);
+  }
+
+  @Test
+  @DisplayName("test all property types")
+  void allTypes() {
+    this.contextRunner
+        .withPropertyValues(
+            "otel.exporter.otlp.enabled=true",
+            "otel.exporter.otlp.timeout=1s",
+            "otel.exporter.otlp.compression=gzip")
+        .run(
+            context -> {
+              ConfigProperties config = getConfig(context);
+              assertThat(config.getString("otel.exporter.otlp.compression")).isEqualTo("gzip");
+              assertThat(config.getBoolean("otel.exporter.otlp.enabled")).isTrue();
+              assertThat(config.getDuration("otel.exporter.otlp.timeout"))
+                  .isEqualByComparingTo(java.time.Duration.ofSeconds(1));
+            });
+  }
+
+  @ParameterizedTest
+  @MethodSource("headerKeys")
+  @DisplayName("should map headers from spring properties")
+  void mapFlatHeaders(String key) {
+    this.contextRunner
+        .withSystemProperties(key + "=a=1,b=2")
+        .run(
+            context ->
+                assertThat(getConfig(context).getMap(key))
+                    .containsExactly(entry("a", "1"), entry("b", "2")));
+  }
+
+  @ParameterizedTest
+  @MethodSource("headerKeys")
+  @DisplayName("should map headers from spring application.yaml")
+  void mapObjectHeaders(String key) {
+    this.contextRunner
+        .withPropertyValues(key + ".a=1", key + ".b=2")
+        .run(
+            context ->
+                assertThat(getConfig(context).getMap(key))
+                    .containsExactly(entry("a", "1"), entry("b", "2")));
+  }
+
+  public static Stream<Arguments> listProperties() {
+    return Arrays.stream(
+            new String[] {
+              "otel.experimental.metrics.view.config",
+              "otel.experimental.resource.disabled.keys",
+              "otel.propagators",
+              "otel.logs.exporter",
+              "otel.metrics.exporter",
+              "otel.traces.exporter",
+              "otel.instrumentation.http.client.capture-request-headers",
+              "otel.instrumentation.http.client.capture-response-headers",
+              "otel.instrumentation.http.server.capture-request-headers",
+              "otel.instrumentation.http.server.capture-response-headers",
+              "otel.instrumentation.http.known-methods",
+            })
+        .map(Arguments::of);
+  }
+
+  @ParameterizedTest
+  @MethodSource("listProperties")
+  @DisplayName("should map list from application.yaml list")
+  void listsShouldWorkWithYaml(String key) {
+    List<String> expected;
+    if (key.equals("otel.propagators")) {
```

Comment:
The test could be easier to read by removing logic from the test.

Reply:
fixed